### DISCUSSION TITLE
multipart/form-data help

### DISCUSSION THREAD

**Discussion by NaviTheCoderboi** [elysiajs/elysia#676] [Q&A]
guys i have this elysia model
```ts
export const createSongModel = new Elysia().model({
    createSongModel: t.Object({
        title: t.String(),
        artist: t.String(),
        artistSocial: t.Optional(t.String()),
        album: t.String(),
        playlistId: t.Number(),
        song: t.File({ type: 'audio/mpeg' })
    })
});
```
```ts
{
            type: 'multipart/form-data',
            body: 'createSongModel',
            detail: {
                tags: ['Song'],
                summary: 'Create a song'
            }
        }
```
why do i get error even i am sending formData
![image](https://github.com/elysiajs/elysia/assets/102732282/2646ba2b-8573-44af-8979-455bb5d746ba)

**Comment by yigsvnsla** [comment]
+1

**Comment by mjjabarullah** [comment]
@NaviTheCoderboi , @yigsvnsla replace type: 'multipart/form-data' with parse: 'multipart/form-data'. this enforces parsing the body as formdata

**Comment by VincentBai-dotcom** [comment]
It's possibly due to multipart/form-data only allows transmission of files or strings. When you click "Send", the playlistId will get converted to String before sending to your server, which triggers validation error from Elysia. To solve the issue, set the type of this field to t.Numeric() : https://elysiajs.com/patterns/typebox.html#numeric
